stomp_plugins/src/cost_functions/cartesian_distance.cpp

```diff
@@ -0,0 +1,182 @@
+    name_("CartesianDistance")
+{
+  // TODO Auto-generated constructor stub
+
+}
+
+CartesianDistance::~CartesianDistance()
+{
+  // TODO Auto-generated destructor stub
+}
+
+bool CartesianDistance::initialize(moveit::core::RobotModelConstPtr robot_model_ptr,
+                        const std::string& group_name,XmlRpc::XmlRpcValue& config)
+{
+  group_name_ = group_name;
+  robot_model_ = robot_model_ptr;
+
+  return configure(config);
+}
+
+bool CartesianDistance::configure(const XmlRpc::XmlRpcValue& config)
+{
+  using namespace XmlRpc;
+
+  try
+  {
+    XmlRpcValue params = config;
+
+    position_cost_weight_ = static_cast<double>(params["position_cost_weight"]);
+    orientation_cost_weight_ = static_cast<double>(params["orientation_cost_weight"]);
+    translation_tolerance_ =  static_cast<double>(params["translation_tolerance"]);
+    rotation_tolerance_ =  static_cast<double>(params["rotation_tolerance"]);
+
+    if (translation_tolerance_ <= 0)
+    {
+      ROS_ERROR("%s the 'translation_tolerance' parameter must not have non positive value",getName().c_str());
+      return false;
+    }
+
+    if (rotation_tolerance_ <= 0)
+    {
+      ROS_ERROR("%s the 'rotation_tolerance_' parameter must not have non positive value",getName().c_str());
+      return false;
+    }
+
+    // total weight
+    cost_weight_ = position_cost_weight_ + orientation_cost_weight_;
+
+  }
+  catch(XmlRpc::XmlRpcException& e)
+  {
+    ROS_ERROR("%s failed to load parameters, %s",getName().c_str(),e.getMessage().c_str());
+    return false;
+  }
+
+  return true;
+}
+
+bool CartesianDistance::setMotionPlanRequest(const planning_scene::PlanningSceneConstPtr& planning_scene,
+                 const moveit_msgs::MotionPlanRequest &req,
+                 const stomp_core::StompConfiguration &config,
+                 moveit_msgs::MoveItErrorCodes& error_code)
+{
+  using namespace moveit::core;
+
+  const JointModelGroup* joint_group = robot_model_->getJointModelGroup(group_name_);
+  int num_joints = joint_group->getActiveJointModels().size();
+  tool_link_ = joint_group->getLinkModelNames().back();
+  state_.reset(new RobotState(robot_model_));
+  robotStateMsgToRobotState(req.start_state,*state_);
+
+  const std::vector<moveit_msgs::Constraints>& seed = req.trajectory_constraints.constraints;
+  if(seed.empty())
+  {
+    ROS_ERROR("A seed trajectory was not provided");
+    error_code.val = error_code.INVALID_GOAL_CONSTRAINTS;
+    initial_trajectory_.clear();
+    return false;
+  }
+
+  initial_trajectory_.resize(seed.size());
+  for (size_t i = 0; i < seed.size(); ++i) {
+    // Test the first point to ensure that it has all of the joints required
+    auto n = seed.at(i).joint_constraints.size();
+    if (n != num_joints)
+    {
+      ROS_ERROR("Seed trajectory index %lu does not have %lu constraints (has %lu instead).", i, num_joints, n);
+      error_code.val = error_code.INVALID_GOAL_CONSTRAINTS;
+      initial_trajectory_.clear();
```

Comment:
I don't like how `initial_trajectory_` member variable is shared between 2 functions. It look like it can not be avoided because of virtual interface but i would recommend you to use local variable and assign / mode to class only and only when it is valid. Also clear on the beginning of `setMotionPlanRequest` would be appropriate (make it as simple as ossible).